src/core/qgsrendercontext.h

```diff
@@ -151,8 +176,20 @@ class CORE_EXPORT QgsRenderContext : public QgsTemporalRangeObject
      * \see setDisabledSymbolLayers()
      * \see isSymbolLayerEnabled()
      * \since QGIS 3.12
+     * \deprecated since QGIS 3.30 and replaced with disabledSymbolLayersV2
+     */
+    Q_DECL_DEPRECATED QSet<const QgsSymbolLayer *> disabledSymbolLayers() const;
+
+    /**
+     * When rendering a map layer in a second pass (for selective masking),
+     * some symbol layers may be disabled.
+     *
+     * Returns the list of disabled symbol layer ids.
+     * \see setDisabledSymbolLayers()
+     * \see isSymbolLayerEnabled()
+     * \since QGIS 3.30
      */
-    QSet<const QgsSymbolLayer *> disabledSymbolLayers() const { return mDisabledSymbolLayers; }
+    const QSet<QString> &disabledSymbolLayersV2() const;
```

Comment:
```suggestion
    QSet<QString> disabledSymbolLayersV2() const;
```

Since QSet is implicitly shared, there's no advantage in return a const ref (and doing so causes potential issues in sip/python code)